applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -425,19 +564,149 @@ static void endpoint_cb(struct bt_conn *conn, enum bt_audio_dir dir, struct bt_b
 		return;
 	}
 
-	if (IS_ENABLED(CONFIG_BT_AUDIO_RX)) {
-		if (dir == BT_AUDIO_DIR_SOURCE) {
-			if (ep != NULL) {
-				headsets[channel_index].source_ep = ep;
-				return;
+	if (dir == BT_AUDIO_DIR_SOURCE) {
+		if (ep != NULL) {
+			headsets[channel_index].source_ep = ep;
+			return;
+		}
+
+		if (headsets[channel_index].source_ep == NULL) {
+			LOG_WRN("No source endpoints found");
+			return;
+		}
+	} else {
+		LOG_WRN("Endpoint direction not recognized: %d", dir);
+	}
+}
+
+static void discover_cb(struct bt_conn *conn, int err, enum bt_audio_dir dir)
+{
+	int ret;
+	uint8_t channel_index = 0;
+	uint8_t temp_cap_index;
+
+	ret = channel_index_get(conn, &channel_index);
+	if (ret) {
+		LOG_ERR("Unknown connection, should not reach here");
+		return;
+	}
+
+	if (err == BT_ATT_ERR_ATTRIBUTE_NOT_FOUND) {
+		if (dir == BT_AUDIO_DIR_SINK) {
```

Comment:
Is this callback called twice if the discover both source and sink?

Reply:
When you call discover you have to specify what direction to discover, so this callback will be called for each call to discover that we have.